app/controllers/holidays_controller.rb

```diff
@@ -84,16 +130,28 @@ def holiday_params
   end
 
   def calculate_leave(operator, length)
-    length_last_year = (current_user.remaining_leave_last_year - length > 0) ? (current_user.remaining_leave_last_year - length) : current_user.remaining_leave_last_year
-    current_user.update_attribute(:remaining_leave_last_year, current_user.remaining_leave_last_year.send(operator, length_last_year))
-    current_user.update_attribute(:remaining_leave, current_user.remaining_leave.send(operator, length))
+    length_last_year = (@holiday.user.remaining_leave_last_year.send(operator, length) > 0) ? (@holiday.user.remaining_leave_last_year.send(operator, length)) : 0
+    @holiday.user.update_attribute(:remaining_leave_last_year, length_last_year)
+    @holiday.user.update_attribute(:remaining_leave, @holiday.user.remaining_leave.send(operator, length))
   end
 
   def add_leave(length)
-    calculate_leave(:+, length)
+    updated_remaining_leave_last_year = @holiday.user.remaining_leave_last_year + @holiday.length_last_year
+    user = @holiday.user
+    user.update_attributes(remaining_leave_last_year: updated_remaining_leave_last_year, remaining_leave: @holiday.user.remaining_leave + length)
+    user.valid?
```

Comment:
In reject prüfe ich, ob das funktioniert hat. Aber an der Fehlermeldung kann man sehen, dass ich nicht wirklich weiß, was da falsch laufen könnte. Kann ich also wohl rausnehmen.